### DISCUSSION TITLE
apply photocopy effect for image ?

### DISCUSSION THREAD

**Discussion by KaitoVux** [ImageMagick/ImageMagick#3652] [Digital Image Processing]
Can I apply some method like the photocopy effect of photoshop to my image? Please help me. Thanks a lot. 
I use IM version 6.0.

![200](https://user-images.githubusercontent.com/39855103/117575763-72952400-b10d-11eb-8a47-1f68d519b519.png)
https://user-images.githubusercontent.com/39855103/117575763-72952400-b10d-11eb-8a47-1f68d519b519.png

The result is:
![200](https://user-images.githubusercontent.com/39855103/117575801-96f10080-b10d-11eb-8af0-eebb1e3216a0.jpg)
https://user-images.githubusercontent.com/39855103/117575801-96f10080-b10d-11eb-8af0-eebb1e3216a0.jpg

**Comment by snibgo** [comment]
What version of IM, on what platform?

Please show "before and after" examples of the effect you want.

**Reply by KaitoVux** [reply]
I just updated the examples above. I use Google Colab to do it. Thanks.

**Comment by fmw42** [comment]
ImageMagick does not have a grid warping function. But you can do something similar using displacement mapping.

Input:

![texture](https://user-images.githubusercontent.com/13458218/117546458-627f2680-afdf-11eb-9f2f-f0d0014ef0dc.png).  

Here I read the texture image and brighten it. Then create a text image on transparent background. Then I create a random noise image and blur it. Then I use the blurred random noise image as a displacement map to warp the text image. Finally, I overlay the warped text image onto the input.


```
magick texture.png -level 0x70% \
\( -clone 0 -fill black -colorize 100 -alpha transparent \
    -fill white -font courier -pointsize 256 -gravity center \
    -annotate +0+0 "PHOTO\nCOPY-" \) \
\( -clone 0 -fill black -colorize 100 +noise random \
	-channel g -separate +channel -blur 0x10 \) \
\( -clone 1,2 -compose Displace -define compose:args=150x150 -composite \) \
-delete 1,2 \
-compose over -composite \
bad_photo_copy.png

```

![bad_photo_copy](https://user-images.githubusercontent.com/13458218/117546529-b25ded80-afdf-11eb-8650-98329c5e846c.png)

**Comment by snibgo** [comment]
Most of the effect is edge-finding, with heavy clipping of highlights. Windows BAT syntax:
```
magick ^
  117575763-72952400-b10d-11eb-8a47-1f68d519b519.png ^
  -alpha off ^
  -colorspace Gray ^
  -unsharp 0x5+10+0 ^
  -contrast-stretch 5x80%% ^
  c.png
```
There is also a blur, which I haven't bothered with.
![c](https://user-images.githubusercontent.com/33812368/117578180-08688900-b0e5-11eb-92a2-1f03decfa570.png)

**Reply by datvtzodinet** [reply]
I will try your solution, many thanks. <3

**Comment by fmw42** [comment]
The result looks like a LoG or DoG. So here is one way.

`convert face.png -colorspace gray -bias 100% -define convolve:scale=8\! -morphology convolve DoG:0,2,6 result.png`

![result](https://user-images.githubusercontent.com/13458218/117578406-0e3d7a80-b0a3-11eb-8ed1-bd2aaef920b9.png)

**Reply by datvtzodinet** [reply]
It seems what I'm looking for, thank you.
